packages/framework/tests/Test/Codeception/AdminRadiobutton.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\FrameworkBundle\Test\Codeception;
+
+use PHPUnit\Framework\Assert;
+
+/**
+ * Representation of a graphical radiobutton that is used in administration
+ * Allows to manipulate radiobuttons and read their state
+ * (The original input is hidden by JS and replaced by a graphical element, therefore it cannot be manipulated directly)
+ */
+class AdminRadiobutton
+{
+    /**
+     * @var \Tests\FrameworkBundle\Test\Codeception\ActorInterface
+     */
+    protected $tester;
+
+    /**
+     * @var string
+     */
+    protected $cssSelector;
+
+    /**
+     * @param \Tests\FrameworkBundle\Test\Codeception\ActorInterface $tester
+     * @param string $cssSelector
+     */
+    protected function __construct(ActorInterface $tester, string $cssSelector)
+    {
+        $this->tester = $tester;
+        $this->cssSelector = $cssSelector;
+    }
+
+    /**
+     * @param \Tests\FrameworkBundle\Test\Codeception\ActorInterface $tester
+     * @param string $cssSelector
+     * @return \Tests\FrameworkBundle\Test\Codeception\AdminRadiobutton
+     */
+    public static function createByCss(ActorInterface $tester, string $cssSelector): self
+    {
+        return new static($tester, $cssSelector);
+    }
+
+    /**
+     * @param string $radioValue
+     */
+    public function select(string $radioValue): void
+    {
+        $imageElementClass = $this->getImageElementClassByValue($radioValue);
+
+        $this->tester->clickByCss('.' . $imageElementClass);
+    }
+
+    /**
+     * @param string $radioValue
+     */
+    public function assertSelected(string $radioValue): void
+    {
+        $imageElementClass = $this->getImageElementClassByValue($radioValue);
+
+        $this->tester->canSeeElement(['css' => '.' . $imageElementClass]);
+
+        $script = sprintf('return $(".%s").is("input:checked + *")', $imageElementClass);
```

Comment:
oh, what does this syntax mean? `input:checked + *`

Reply:
It is combination of https://api.jquery.com/next-adjacent-selector/ and https://api.jquery.com/all-selector/

 `sprintf()` is hard to reading for me in this case...

Reply:
Yes, it answers the question whether the element with `$imageElementClass` is directly adjacent to a checked `input`.

The `sprintf` might be a bit hard to read but it was even worse when I've used concatenation - it was pretty easy to miss a quote.